src/Makefile-env.am

```diff
@@ -87,6 +86,18 @@ AM_COMMON_CPPFLAGS = \
 	-DCEPH_PKGLIBDIR=\"${pkglibdir}\" \
 	-DGTEST_USE_OWN_TR1_TUPLE=0
 
+if LINUX
+AM_COMMON_CPPFLAGS += \
+	-D_REENTRANT
+endif
+
+if SOLARIS
+AM_COMMON_CPPFLAGS += \
+       -D_PTHREADS \
+       -D_POSIX_C_SOURCE
+	
+endif
```

Comment:
guess `PTHREAD_CFLAGS` offered by `ACX_PTHREAD` would suffice? and we already have it. and the interesting thing is what i found in m4/acx_pthread.m4 and 
```m4
        case "${host_cpu}-${host_os}" in
            *-aix* | *-freebsd* | *-darwin*) flag="-D_THREAD_SAFE";;
            *solaris* | *-osf* | *-hpux*) flag="-D_REENTRANT";;
        esac
```

http://git.savannah.gnu.org/gitweb/?p=autoconf-archive.git;a=blob_plain;f=m4/ax_pthread.m4 shows something different, but similar
```m4
 AC_MSG_CHECKING([if more special flags are required for pthreads])
        flag=no
        case ${host_os} in
            aix* | freebsd* | darwin*) flag="-D_THREAD_SAFE";;
            osf* | hpux*) flag="-D_REENTRANT";;
            solaris*)
            if test "$GCC" = "yes"; then
                flag="-D_REENTRANT"
            else
                # TODO: What about Clang on Solaris?
                flag="-mt -D_REENTRANT"
            fi
            ;;
        esac
```